libraries/botbuilder-dialogs-adaptive-testing/src/testRunner.ts

```diff
@@ -16,34 +16,33 @@ import { TestScript } from './testScript';
  * In charge of running Dialog Adaptive tests.
  */
 export class TestRunner {
-    private resourceExplorer: ResourceExplorer;
-    private testAdapter: TestAdapter;
-
     /**
      * Initializes a new instance of the `TestRunner` class.
      * @param resourcePath Path where the `.dialog` files are located.
      */
-    public constructor(resourcePath: string) {
+    public constructor(private resourcePath: string) {
         ComponentRegistration.add(new AdaptiveComponentRegistration());
         ComponentRegistration.add(new AdaptiveTestComponentRegistration());
         ComponentRegistration.add(new QnAMakerComponentRegistration());
         ComponentRegistration.add(new LuisComponentRegistration());
-
-        this.resourceExplorer = new ResourceExplorer();
-        this.resourceExplorer.addFolder(resourcePath, true, false);
-
-        this.testAdapter = new TestAdapter(TestAdapter.createConversation('botbuilder-dialogs-adaptive-testing'));
     }
 
     /**
      * Runs a test script with the specified name.
      * @param testName Test name.
      * @returns A Promise that represents the work queued to execute.
      */
-    public async runTestScript(testName: string): Promise<any> {
-        const script = this.resourceExplorer.loadType<TestScript>(`${testName}.test.dialog`);
+    public runTestScript(testName: string): Promise<void> {
+        // Construct a fresh test adapter
+        const testAdapter = new TestAdapter(TestAdapter.createConversation('botbuilder-dialogs-adaptive-testing'));
+
+        // Construct a fresh resourceExplorer
+        const resourceExplorer = new ResourceExplorer();
+        resourceExplorer.addFolder(this.resourcePath, true, false);
```

Comment:
Similar to my earlier comment about the `TestAdapter`, this is unsettling and I'm not a fan of it at the moment.

To support your thought in https://github.com/microsoft/botbuilder-js/pull/2699/files#r508017128, I believe instantiating the ResourceExplorer is expensive, so we should probably setup a performance test.

For this change, we get out of the event listener issue but ultimately `this.resourcePath` is a private member, so recreating it over and over again seems odd.

___

#### Proposed alternative

A `ResourceExplorer` can be passed in to the constructor, which renders `this.resourcePath` moot. If they pass in a ResourceExplorer we set it on a private property and use that for every `runTestScript()` call.

If they pass in a `resourcePath: string` to the constructor we can either arrange a new ResourceExplorer for every `runTestScript()` call or create and populate a ResourceExplorer for the same private property.

Reply:
> Proposed alternative

I think this might get us back to square 1 with the same issue of exceeding event listeners. It might mitigate it, some, since there's some instances where we're re-creating the `ResourceExplorer`, but I could see us running into it again in the future.

Reply:
@stevengum it's on the order of 2 milliseconds, so definitely not a real concern IMO.